File: BOJ/Calendar/Kotlin/3183.kt
import java.util.Scanner

fun isLeafYear(year: Int) : Boolean {
	return (year % 400 == 0) || (year % 100 != 0 && year % 4 == 0) 
}

fun isValid(day: Int, month: Int, year: Int) : Boolean {
	var days = Array(13) { 31 }

	for (month in listOf(4, 6, 9, 11))
		days[month] = 30
	
	days[2] = if (isLeafYear(year)) 29 else 28
	
	return month >= 1 && month <= 12 && day >= 1 && day <= days[month]
}

fun main(args: Array<String>) = with(Scanner(System.`in`)) {
	while(true) {
		val day = nextInt()
		val month = nextInt()
		val year = nextInt()
		
		if (day == 0 && month == 0 && year == 0)
			break
		
		println(if (isValid(day,month,year)) "Valid" else "Invalid")
	}
}

File: BOJ/Calendar/Python/3183.py
def is_leaf_year(year):
	return (year % 400 == 0) or (year % 100 != 0 and year % 4 == 0)

def is_valid(day, month, year):
	days = [31, 29 if is_leaf_year(year) else 28, 31,
			30, 31, 30, 31,
			30, 31, 30, 31]
	
	return day >= 1 and day <= days[month-1] and month >= 1 and month <= 12 

while(True):
	day,month,year=map(int, input().split())
	if day == 0 and month == 0 and year == 0:
		break
	print("Valid" if is_valid(day,month,year) else "Invalid")
